File: Open_Weather_API_App.py
Sure, here is a simple Python script that uses the Open Weather API:

```python
import requests
import json

def get_weather(api_key, city):
    base_url = "http://api.openweathermap.org/data/2.5/weather"
    params = {
        'q': city,
        'appid': api_key,
        'units': 'metric'
    }
    response = requests.get(base_url, params=params)
    return response.json()

def print_weather(data):
    print(f"City: {data['name']}")
    print(f"Weather: {data['weather'][0]['description']}")
    print(f"Temperature: {data['main']['temp']}°C")
    print(f"Humidity: {data['main']['humidity']}%")
    print(f"Wind: {data['wind']['speed']} m/s")

def main():
    api_key = "your_api_key_here"
    city = input("Enter city name: ")
    weather_data = get_weather(api_key, city)
    print_weather(weather_data)

if __name__ == "__main__":
    main()
```

Please replace "your_api_key_here" with your actual Open Weather API key. This script will prompt the user to enter a city name, then it will display the current weather information for that city.

File: Open_Weather_API_App.java
Here is a simple Java console application that uses the Open Weather API:

```java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class Main {
    private static final String API_KEY = "YOUR_API_KEY";
    private static final String API_URL = "http://api.openweathermap.org/data/2.5/weather?q=";

    public static void main(String[] args) throws Exception {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Enter city name: ");
        String city = reader.readLine();
        String urlRequest = API_URL + city + "&appid=" + API_KEY;
        URL url = new URL(urlRequest);
        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setRequestMethod("GET");
        BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
        String inputLine;
        StringBuffer content = new StringBuffer();
        while ((inputLine = in.readLine()) != null) {
            content.append(inputLine);
        }
        in.close();
        connection.disconnect();
        System.out.println("Weather Data: " + content.toString());
    }
}
```

Please replace "YOUR_API_KEY" with your actual Open Weather API key. This application will prompt the user to enter a city name, then it will fetch and display the weather data for that city.